modules/RegeneronGeneBurden.py

```diff
@@ -0,0 +1,326 @@
+    "M1.0001": "Burden test carried out with pLOF variants with a MAF smaller than 0.001%.",
+    "M1.001": "Burden test carried out with pLOF variants with a MAF smaller than 0.01%.",
+    "M1.01": "Burden test carried out with pLOF variants with a MAF smaller than 0.1%.",
+    "M1.1": "Burden test carried out with pLOF variants with a MAF smaller than 1%.",
+    "M3.singleton": "Burden test carried out with singleton pLOF and deleterious missense variants.",
+    "M3.0001": "Burden test carried out with pLOF and deleterious missense variants with a MAF smaller than 0.001%.",
+    "M3.001": "Burden test carried out with pLOF and deleterious missense variants with a MAF smaller than 0.01%.",
+    "M3.01": "Burden test carried out with pLOF and deleterious missense variants with a MAF smaller than 0.1%.",
+    "M3.1": "Burden test carried out with pLOFs and deleterious missense variants with a MAF smaller than 1%.",
+}
+
+
+def main(regeneron_data: str, gwas_studies: str, spark_instance: SparkSession) -> DataFrame:
+    """
+    This module extracts and processes target/disease evidence from the raw data published in PMID:34662886.
+    Args:
+    """
+    logging.info(f"File with the results from the REGENERON burden analyses: {regeneron_data}")
+    logging.info(f"File with GWAS Catalog studies: {gwas_studies}")
+
+    # Load data
+
+    TO_KEEP = [
+        "Gene",
+        "Trait",
+        "Trait type",
+        "UKB detailed trait name",
+        "Trait_String",
+        "Marker",
+        "Marker type",
+        "P-value",
+        "Effect (95% CI)",
+        "N cases with 0|1|2 copies of effect allele",
+        "N controls with 0|1|2 copies of effect allele",
+        "Ancestry",
+    ]
+
+    eur_evd_df = (
+        spark_instance.createDataFrame(
+            read_excel(regeneron_data, sheet_name="SD2").filter(items=TO_KEEP).replace({nan: None}).astype(str)
+        )
+        .withColumn("Ancestry", lit("EUR"))
+        .withColumnRenamed("UKB detailed trait name", "Study tag")
+    )
+
+    non_eur_evd_df = spark_instance.createDataFrame(
+        read_excel(regeneron_data, sheet_name="SD3").filter(items=TO_KEEP).replace({nan: None}).astype(str)
+    ).withColumnRenamed("Trait_String", "Study tag")
+
+    summary_stats_df = spark_instance.createDataFrame(
+        read_excel(regeneron_data, sheet_name="SD4", skiprows=0, header=1)
+        .iloc[2:]
+        .reset_index(drop=True)
+        .filter(items=["Study tag", "Study Accession"])
+        .replace({nan: None})
+        .astype(str)
+    ).withColumn("Study tag", split("Study tag", "__SV").getItem(0))
+
+    gwas_studies_df = (
+        spark_instance.read.csv(gwas_studies, header=True, inferSchema=True, sep="\t")
+        .filter(col("LINK").contains("34662886"))
+        .select(
+            col("STUDY ACCESSION").alias("Study Accession"),
+            element_at(split("MAPPED_TRAIT_URI", "/"), -1).alias("MAPPED_TRAIT"),
```

Comment:
This is not good. Some studies are assigned multiple EFO ids by the GWAS Catalog. In such cases the list of IDs are separated by commas. The above solution only returns the id of the last one. However there are 104 studies in this publication such studies. eg:

```
-RECORD 0------------------------------------------------------------------------------------------------
 DATE ADDED TO CATALOG      | 2022-03-01                                                                 
 PUBMEDID                   | 34662886                                                                   
 FIRST AUTHOR               | Backman JD                                                                 
 DATE                       | 2021-10-18                                                                 
 JOURNAL                    | Nature                                                                     
 LINK                       | www.ncbi.nlm.nih.gov/pubmed/34662886                                       
 STUDY                      | Exome sequencing and analysis of 454,787 UK Biobank participants.          
 DISEASE/TRAIT              | Venous thromboembolism - composite (Gene-based burden)                     
 INITIAL SAMPLE SIZE        | 5,702 European ancestry cases, 220,211 European ancestry controls          
 REPLICATION SAMPLE SIZE    | NA                                                                         
 PLATFORM [SNPS PASSING QC] | Illumina [180656]                                                          
 ASSOCIATION COUNT          | 0                                                                          
 MAPPED_TRAIT               | venous thromboembolism, deep vein thrombosis                               
 MAPPED_TRAIT_URI           | http://www.ebi.ac.uk/efo/EFO_0004286, http://www.ebi.ac.uk/efo/EFO_0003907 
 STUDY ACCESSION            | GCST90085520                                                               
 GENOTYPING TECHNOLOGY      | Exome-wide sequencing                                                      
only showing top 1 row
```
The script needs to be equipped to handle list of EFOs.

This might work:
```python
expr("regexp_extract_all(MAPPED_TRAIT_URI, '([A-Z]+_[0-9]+)')").alias('MAPPED_TRAIT')
```